backend/src/main/java/com/htecgroup/skynest/model/dto/FolderDto.java

```diff
@@ -26,4 +24,13 @@ public class FolderDto {
   @With private FolderDto parentFolder;
 
   @With private BucketDto bucket;
+
+  public FolderDto deleteFolder() {
+    return this.withDeletedOn(LocalDateTime.now());
+  }
+
+  public boolean isDeleted() {
+    if (deletedOn != null) return true;
```

Comment:
Why not just `return deletedOn != null`

Reply:
I left it like this on purpose for you to catch it, nice job !